### PR TITLE
[wasm] Fix WebSocket tests on NodeJS

### LINKED ISSUE #64473: NodeJS packed as correlation payload doesn't preserve symbolic links
NodeJS packed as correlation payload doesn't preserve symbolic links. Instead it copies target files to places where links sit.

**node/14.15.5_64bit should contain**
```
npm -> ../lib/node_modules/npm/bin/npm-cli.js*
npx -> ../lib/node_modules/npm/bin/npx-cli.js*
```

- [Workitem](https://helix.dot.net/api/jobs/5afc8e6e-d93d-42e6-818a-d467d10d5a65/workitems/WasmTestOnNodeJs-System.Net.WebSockets.Client.Tests?api-version=2019-06-17)
- [Correlation payload](https://helixde8s23ayyeko0k025g8.blob.core.windows.net/helix-job-b0993f53-996e-4167-9ded-4876855cb240b73c9ff8f5649fb9b/05e80c2e-09f1-4b4b-b278-f77d9c106244.zip)

Running `npm` could cause errors like (https://github.com/dotnet/runtime/pull/64330)

```
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module '../lib/utils/unsupported.js'
Require stack:
- /datadisks/disk1/work/BA3509B2/p/build/emsdk/node/14.15.5_64bit/bin/npm
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at /datadisks/disk1/work/BA3509B2/p/build/emsdk/node/14.15.5_64bit/bin/npm:19:21
    at Object.<anonymous> (/datadisks/disk1/work/BA3509B2/p/build/emsdk/node/14.15.5_64bit/bin/npm:155:3)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/datadisks/disk1/work/BA3509B2/p/build/emsdk/node/14.15.5_64bit/bin/npm'
  ]
}
```

### Possible solutions
- Use cli zip to pack the nodejs with correct symlinks
- ~Install/extract nodejs on Helix using emsdk installation script~ (it uses python to download and extract it).
- Download and extract nodejs from official site, like emsdk does it.
- Install nodejs on Helix VM image
- ~Use nodejs from emsdk nuget package~ (it is stripped, eg npm is missing)

### PR DESCRIPTION
**Pull request by maraf**
- Fixes https://github.com/dotnet/runtime/issues/64473 - restore two symbolic links.
- Add `WasmXHarnessArgsCli` as an alternative variable that can be used from cli, as `WasmXHarnessMonoArgs` is used in some csprojs.
- Include `TestEchoMiddleware` and `RemoteLoopMiddleware` as correlation payload even for NodeJS.
- Move `SetScriptCommands` to a targets so they are in correct order in the generated run script.

### CODE REVIEW

**Comment by pavelsavara** [comment]
Perhaps unrelated, but why we have [`WasmXHarnessArgs` into `_XHarnessArgs` ](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/eng/testing/tests.wasm.targets#L48)

and then both [`_XHarnessArgs` and `WasmXHarnessArgs` into `RunScriptCommand`](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/eng/testing/tests.wasm.targets#L56)

Does that duplicate `WasmXHarnessArgs` ?

Also, could we improve it so that I could pass something like `/p:WasmXHarnessArgs="--no-headless"` from commandline without ruining [--web-server-use-cors](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/src/libraries/Common/tests/System/Net/Prerequisites/LocalEchoServer.props#L12) by forcing `WasmXHarnessArgs` to constant.

**Comment by maraf** [comment]
> Perhaps unrelated, but why we have [`WasmXHarnessArgs` into `_XHarnessArgs` ](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/eng/testing/tests.wasm.targets#L48)
> 
> and then both [`_XHarnessArgs` and `WasmXHarnessArgs` into `RunScriptCommand`](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/eng/testing/tests.wasm.targets#L56)

There are two flavors of those - MSBuild property vs environment variable. I have added a comment about it.

> Also, could we improve it so that I could pass something like `/p:WasmXHarnessArgs="--no-headless"` from commandline without ruining [--web-server-use-cors](https://github.com/dotnet/runtime/blob/3361b2f86acdfb9095ecc0e2e4aea539d8fb72cb/src/libraries/Common/tests/System/Net/Prerequisites/LocalEchoServer.props#L12) by forcing `WasmXHarnessArgs` to constant.

I have add a new MSBuild property `WasmXHarnessArgsCli` that behaves as `WasmXHarnessArgs`, but it's used in csprojs.

File: src/libraries/sendtohelix-wasm.targets [hunk 1]
```diff
@@ -165,6 +164,19 @@
     </PropertyGroup>
 
     <Message Condition="'$(NeedsEMSDK)' == 'true' or '$(NeedsEMSDKNode)' == 'true'" Importance="High" Text="Using emsdk: $(EmSdkDirForHelixPayload)" />
+    
+    <ItemGroup Condition="'$(EnableDefaultBuildHelixWorkItems)' == 'true'">
```
Comment (radical) [inline src/libraries/sendtohelix-wasm.targets:168]:
```suggestion
    <ItemGroup Condition="'$(IsRunningLibraryTests)' == 'true'">
```

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**